components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -302,3 +302,13 @@ sealed class EngineAction : BrowserAction() {
         val engineSessionState: EngineSessionState
     ) : EngineAction()
 }
+
+/**
+ * [BrowserAction] implementations relating to update the completed DownloadState inside [BrowserState]
+ */
+sealed class DownloadAction : BrowserAction() {
+    /**
+     * Adds the [downloadId] into the set of completed downloads inside the [BrowserState].
+     */
+    data class DownloadCompletedAction(val downloadId: Long) : DownloadAction()
```

Comment:
I wonder whether we could put the whole `Download` object into this action. But I assume this does not work in the case where we pass the download to Android's Download Manager (except if we keep the download object in the download manager wrapper?).